.circleci/config.yml

```diff
@@ -33,7 +33,7 @@ default_environment: &default_environment
 executors:
   golang:
     docker:
-      - image: circleci/golang:1.14.4
+      - image: circleci/golang:1.15.8
```

Comment:
It's a bit awkward that we track this here, below, and in the orb https://github.com/ipfs/ci-helpers/blob/fb84aaad053f46509acd7264420a57629ec95046/orbs/ci-go/%40root.yml#L26. Does this have to be duplicated?

Reply:
I think we _only_ set it here to override the orb's default. We can update the one in the orb and delete this if you'd like.

Reply:
Looks like you may also want to change line 63 to have the same version of go.

At this point, probably want to update to test with `go1.16.2`

Reply:
I switched everything to 1.15.2 since that's what CircleCI gives us for the sharness containers and I figured it'd be good to keep all the testing aligned for now. We should figure out running tests for Go 1.15 and 1.16 (i.e. latest and latest-1)